src/compressor/CMakeLists.txt

```diff
@@ -22,10 +29,7 @@ if(HAVE_BROTLI)
   add_subdirectory(brotli)
 endif()
 
-add_library(compressor STATIC ${compressor_srcs})
-if(HAVE_QATZIP)
-  target_link_libraries(compressor PRIVATE ${QATZIP_LIBRARIES})
-endif()
+add_library(compressor STATIC $<TARGET_OBJECTS:compressor_objs>)
```

Comment:
We've removed the redundant part, and use `add_library` to create target `compressor`. 
There are already two libraries: the Object library `compressor_objs` which has been used to keep build directories less cluttered and speed up the build process, and the library `compressor` invoked by compressor libraries.
 @tchaikov